### ISSUE TITLE
Build problems on Ubuntu 16.04

### ISSUE THREAD

**Issue by IanHeywood** [ska-sa/meqtrees-timba#36]
Trying to build MeqTrees in my home area on a Ubuntu 16.04 machine that I don't own, and getting the following mess. As far as I'm aware the dependencies are all available.

Any help deciphering this will be gratefully received.

Cheers.

```
ianh@pi1:~/Software/MeqTrees/meqtrees-timba$ Tools/Build/bootstrap_cmake release
Configuring release
executing cmake -DCMAKE_BUILD_TYPE=RELEASE /home/ianh/Software/MeqTrees/meqtrees-timba
-- The CXX compiler identification is GNU 5.4.0
-- The C compiler identification is GNU 5.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
CUDA_TOOLKIT_ROOT_DIR not found or specified
-- Could NOT find CUDA (missing:  CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY) (Required is at least version "4.0")
 * CUDA not found -- CUDA support disabled. This support is still only experimental, do not be alarmed.
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for include file malloc.h
-- Looking for include file malloc.h - found
-- Looking for include files sys/param.h, sys/mount.h
-- Looking for include files sys/param.h, sys/mount.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Found PythonInterp: /usr/bin/python2.7 (found version "2.7.12")
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found version "2.7.12")
-- INSIDE_VIRTUALENV = False
-- PYTHON_PACKAGES_DIR = /usr/local/lib/python2.7/dist-packages
-- Found BLITZ: /usr/lib/libblitz.so
-- A library with BLAS API found.
-- A library with LAPACK API found.
-- Found CASACORE: /usr/lib/x86_64-linux-gnu/libcasa_images.so;/usr/lib/x86_64-linux-gnu/libcasa_mirlib.so;/usr/lib/libcasa_components.so;/usr/lib/x86_64-linux-gnu/libcasa_coordinates.so;/usr/lib/x86_64-linux-gnu/libcasa_lattices.so;/usr/lib/x86_64-linux-gnu/libcasa_msfits.so;/usr/lib/x86_64-linux-gnu/libcasa_ms.so;/usr/lib/x86_64-linux-gnu/libcasa_fits.so;/usr/lib/x86_64-linux-gnu/libcasa_measures.so;/usr/lib/x86_64-linux-gnu/libcasa_tables.so;/usr/lib/x86_64-linux-gnu/libcasa_scimath.so;/usr/lib/x86_64-linux-gnu/libcasa_scimath_f.so;/usr/lib/x86_64-linux-gnu/libcasa_casa.so;/usr/lib/x86_64-linux-gnu/libcasa_images.so;/usr/lib/libcasa_calibration.so;/usr/lib/libcasa_msvis.so;/usr/lib/libcasa_synthesis.so;/usr/lib/liblapack.so
-- Found DBM: /usr/lib/x86_64-linux-gnu/libgdbm.so
-- Found WCSLIB: /usr/lib/x86_64-linux-gnu/libwcs.so
-- Found FFTW3: /usr/lib/x86_64-linux-gnu/libfftw3.so
-- Found CFITSIO: /usr/lib/x86_64-linux-gnu/libcfitsio.so
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found PYTHON_NUMPY: /usr/lib/python2.7/dist-packages/numpy/core/include
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ianh/Software/MeqTrees/meqtrees-timba/build/release
build configured at /home/ianh/Software/MeqTrees/meqtrees-timba/build/release
ianh@pi1:~/Software/MeqTrees/meqtrees-timba$ cd build/
ianh@pi1:~/Software/MeqTrees/meqtrees-timba/build$ ls
release
ianh@pi1:~/Software/MeqTrees/meqtrees-timba/build$ cd release
ianh@pi1:~/Software/MeqTrees/meqtrees-timba/build/release$ make -j 4
Scanning dependencies of target viscube
Scanning dependencies of target timbase
[  0%] Generating AID-Global-Registry.cc
Scanning dependencies of target octopussy
=== wrote 798 registry entries to /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/DMI/AID-Global-Registry.cc
Scanning dependencies of target dmi
[  0%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/AipsppMutex.cc.o
[  0%] Building CXX object VisCube/CMakeFiles/viscube.dir/src/AID-VisCube-Registry.cc.o
[  1%] Building CXX object DMI/CMakeFiles/dmi.dir/src/AID-DMI-Registry.cc.o
[  1%] Building CXX object OCTOPUSSY/CMakeFiles/octopussy.dir/src/AID-OCTOPUSSY-Registry.cc.o
In file included from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/Lorrays-Blitz.h:1:0,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Lorrays.h:41,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/Lorrays.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TypeId.h:30,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TypeId.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TID-DMI.h:31,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TID-DMI.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TypeInfo.h:29,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TypeInfo.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/VisCube/src/AID-VisCube-Registry.cc:29:
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Lorrays-Blitz.h:46:36: fatal error: casa/Arrays/IPosition.h: No such file or directory
compilation terminated.
VisCube/CMakeFiles/viscube.dir/build.make:62: recipe for target 'VisCube/CMakeFiles/viscube.dir/src/AID-VisCube-Registry.cc.o' failed
make[2]: *** [VisCube/CMakeFiles/viscube.dir/src/AID-VisCube-Registry.cc.o] Error 1
CMakeFiles/Makefile2:429: recipe for target 'VisCube/CMakeFiles/viscube.dir/all' failed
make[1]: *** [VisCube/CMakeFiles/viscube.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  2%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Debug.cc.o
In file included from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/Lorrays-Blitz.h:1:0,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Lorrays.h:41,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/Lorrays.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TypeId.h:30,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TypeId.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TID-DMI.h:31,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TID-DMI.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TypeInfo.h:29,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TypeInfo.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/AID-DMI-Registry.cc:29:
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Lorrays-Blitz.h:46:36: fatal error: casa/Arrays/IPosition.h: No such file or directory
compilation terminated.
In file included from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/Lorrays-Blitz.h:1:0,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Lorrays.h:41,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/Lorrays.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TypeId.h:30,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TypeId.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TID-DMI.h:31,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TID-DMI.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/DMI/src/TypeInfo.h:29,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/DMI/TypeInfo.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/OCTOPUSSY/src/AID-OCTOPUSSY-Registry.cc:29:
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Lorrays-Blitz.h:46:36: fatal error: casa/Arrays/IPosition.h: No such file or directory
compilation terminated.
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Debug.cc:30:0: warning: "MAKE_LOFAR_SYMBOLS_GLOBAL" redefined
 #define MAKE_LOFAR_SYMBOLS_GLOBAL
 ^
<command-line>:0:0: note: this is the location of the previous definition
DMI/CMakeFiles/dmi.dir/build.make:67: recipe for target 'DMI/CMakeFiles/dmi.dir/src/AID-DMI-Registry.cc.o' failed
make[2]: *** [DMI/CMakeFiles/dmi.dir/src/AID-DMI-Registry.cc.o] Error 1
CMakeFiles/Makefile2:204: recipe for target 'DMI/CMakeFiles/dmi.dir/all' failed
make[1]: *** [DMI/CMakeFiles/dmi.dir/all] Error 2
OCTOPUSSY/CMakeFiles/octopussy.dir/build.make:62: recipe for target 'OCTOPUSSY/CMakeFiles/octopussy.dir/src/AID-OCTOPUSSY-Registry.cc.o' failed
make[2]: *** [OCTOPUSSY/CMakeFiles/octopussy.dir/src/AID-OCTOPUSSY-Registry.cc.o] Error 1
CMakeFiles/Makefile2:259: recipe for target 'OCTOPUSSY/CMakeFiles/octopussy.dir/all' failed
make[1]: *** [OCTOPUSSY/CMakeFiles/octopussy.dir/all] Error 2
[  2%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Exception.cc.o
[  2%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/LofarLogCout.cc.o
[  3%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/LofarLogger.cc.o
In file included from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/StringUtil.h:1:0,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/LofarLogger.h:29,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/LofarLogger.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/LofarLogger.cc:28:
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/StringUtil.h:76:36: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
         char*      format = "%F %T");
                                    ^
[  3%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Stopwatch.cc.o
[  4%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/StringUtil.cc.o
In file included from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/StringUtil.h:1:0,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/StringUtil.cc:24:
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/StringUtil.h:76:36: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
         char*      format = "%F %T");
                                    ^
[  4%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Timer.cc.o
[  4%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Net/Socket.cc.o
[  5%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Profiling/PerfProfile.cc.o
[  5%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Thread/SmartLock.cc.o
[  5%] Building CXX object TimBase/CMakeFiles/timbase.dir/src/Thread/Thread.cc.o
In file included from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/StringUtil.h:1:0,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/LofarLogger.h:29,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/build/release/include/TimBase/LofarLogger.h:1,
                 from /home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/Net/Socket.cc:25:
/home/ianh/Software/MeqTrees/meqtrees-timba/TimBase/src/StringUtil.h:76:36: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
         char*      format = "%F %T");
                                    ^
[  6%] Linking CXX shared library libtimbase.so
[  6%] Built target timbase
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
ianh@pi1:~/Software/MeqTrees/meqtrees-timba/build/release$
```

**Comment by IanHeywood** [comment]
From the PI cluster sysadmin...

```
        It couldn't find the casacore include files. I don't know why, they're in a pretty obvious place.
        If you run the config script like this:
        ianh@pi1:~/Software/MeqTrees/meqtrees-timba$ Tools/Build/bootstrap_cmake release -DCASACORE_INCLUDE_DIR=/usr/include/
        Then it looks in the right place for the include files.
        I've run "make" and it's finished ok now. I'll leave the "install" bit up to you...
```

**Comment by o-smirnov** [comment]
Yeah bit of a mystery. /usr/include is bog-standard for casacore, why would it not have found it? I can only venture that there's something funky on the system (CASA itself in an include path?) which it found first and mistook for casacore.

**Comment by IanHeywood** [comment]
Yeah, there is something fishy going on. I was careful not to have CASA involved at any point, having fallen foul of that many times.

There's a system-wide installation of MeqTrees which seems to be broken, hence me rebuilding it for myself on this machine. When I ran timba-init.sh and timba-setup is completely butchered my PATH and PYTHONPATH, featuring loads of places associated with the old system build.

I hardwired the two paths in .bashrc and commented out:

https://github.com/ska-sa/meqtrees-timba/blob/master/install/timba-init.sh#L100-L101

which seemed to do the trick.

(Having ~/Purr in there has always seemed like a bit of a bold assumption...)
